src/ProjectOrigin.Registry.IntegrationTests/ThroughputTests.cs

```diff
@@ -0,0 +1,163 @@
+using System.Threading.Tasks;
+using System;
+using ProjectOrigin.PedersenCommitment;
+using ProjectOrigin.Electricity.V1;
+using ProjectOrigin.HierarchicalDeterministicKeys;
+using System.Diagnostics;
+using ProjectOrigin.HierarchicalDeterministicKeys.Interfaces;
+using DotNet.Testcontainers.Builders;
+using DotNet.Testcontainers.Containers;
+using System.Text;
+using Grpc.Net.Client;
+using DotNet.Testcontainers.Images;
+using System.Collections.Concurrent;
+
+namespace ProjectOrigin.Electricity.IntegrationTests;
+
+public class ThroughputTests : IAsyncLifetime
+{
+    private const string ElectricityVerifierImage = "ghcr.io/project-origin/electricity-server:0.2.0-rc.17";
+    private const string RegistryImage = "ghcr.io/project-origin/registry-server:0.2.0-rc.17";
+    private const int GrpcPort = 80;
+    private const string IssuerArea = "Narnia";
+    private const string RegistryName = "TheRegistry";
+
+    private readonly IFutureDockerImage _registryImage;
+    private readonly IContainer _verifierContainer;
+    private readonly Lazy<IContainer> _registryContainer;
+    private readonly IPrivateKey _issuerKey;
+
+    public ThroughputTests()
+    {
+        _issuerKey = Algorithms.Ed25519.GenerateNewPrivateKey();
+
+        _verifierContainer = new ContainerBuilder()
+                .WithImage(ElectricityVerifierImage)
+                .WithPortBinding(GrpcPort, true)
+                .WithEnvironment($"Issuers__{IssuerArea}", Convert.ToBase64String(Encoding.UTF8.GetBytes(_issuerKey.PublicKey.ExportPkixText())))
+                .WithWaitStrategy(
+                    Wait.ForUnixContainer()
+                        .UntilPortIsAvailable(GrpcPort)
+                    )
+                .Build();
+
+        _registryImage = new ImageFromDockerfileBuilder()
+            .WithDockerfileDirectory(CommonDirectoryPath.GetSolutionDirectory(), string.Empty)
+            .WithDockerfile("ProjectOrigin.Registry.Server/Dockerfile")
+            .Build();
+
+        _registryContainer = new Lazy<IContainer>(() =>
+        {
+            var verifierUrl = $"http://{_verifierContainer.IpAddress}:{GrpcPort}";
+            return new ContainerBuilder()
+                .WithImage(_registryImage)
+                .WithPortBinding(GrpcPort, true)
+                .WithEnvironment("Verifiers__project_origin.electricity.v1", verifierUrl)
+                .WithEnvironment("RegistryName", RegistryName)
+                .WithEnvironment("IMMUTABLELOG__TYPE", "log")
+                .WithEnvironment("VERIFIABLEEVENTSTORE__BATCHSIZEEXPONENT", "0")
+                .WithWaitStrategy(
+                    Wait.ForUnixContainer()
+                        .UntilPortIsAvailable(GrpcPort)
+                    )
+                .Build();
+        });
+
+    }
+
+    private GrpcChannel CreateRegistryChannel()
+    {
+        return GrpcChannel.ForAddress($"http://{_registryContainer.Value.IpAddress}:{GrpcPort}");
+    }
+
+    [Fact(Skip = "Manual test")]
```

Comment:
Any plans on making this test run as part of the test suite?

Reply:
Currently it is a test that builds and starts the container, and a runs for 5 minutes and calculates the throughput based on this, so it is no a test that is valid or not, but simple measures and outputs the throughput.

Reply:
fixed